OSSM-5917: Add text indicating that imported service requires alias

rh-tokeefe opened the pull request:
… when importAsLocal is set to true

<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): OCP 4.12+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: https://issues.redhat.com/browse/OSSM-5917
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://71852--ocpdocs-pr.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-federation

<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

eoinfennessy reviewed (approved):
LGTM

mkralik3 commented:
LGTM

mkralik3 commented on modules/ossm-federation-config-import.adoc:
```diff
@@ -77,7 +77,7 @@ spec:
   - type: NameSelector
     importAsLocal:
-|Set to `true` to aggregate remote endpoint with local services. When `true`, services will be imported as `<name>.<namespace>.svc.cluster.local`
+|Set to `true` to aggregate remote endpoint with local services. When `true` services will be imported as `<name>.<namespace>.svc.cluster.local`. When 'true' an alias is required. When `false` no alias is required. 
```
```
'true'
```
@rh-tokeefe  probably typo. Apostrophes ('') are used instead of Grave accents (``) so that `true` is not highlighted as the others in that section ( https://71852--ocpdocs-pr.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-federation#ossm-federation-config-import_federation )

rh-tokeefe replied:
Great catch @mkralik3 ! PR has been updated. I will squash the PR after you review and give it LGTM.

mkralik3 commented on modules/ossm-federation-create-export.adoc:
```diff
@@ -11,6 +11,11 @@ You create an `ExportedServiceSet` resource to explicitly declare the services t
 
+[NOTE]
+====
+When you set the `importAsLocal` parameter to 'true' to aggregate the remote endpoint with local services you must use an alias for the service. When you set the parameter `false` no alias is required.
```
Same comment as the previous one.

rh-tokeefe replied:
PR revised.

rh-tokeefe commented:
/label peer-review-needed

skopacz1 commented:
/label peer-review-in-progress

skopacz1 commented on modules/ossm-federation-config-import.adoc:
```diff
@@ -77,7 +77,7 @@ spec:
   - type: NameSelector
     importAsLocal:
-|Set to `true` to aggregate remote endpoint with local services. When `true`, services will be imported as `<name>.<namespace>.svc.cluster.local`
+|Set to `true` to aggregate remote endpoint with local services. When `true` services will be imported as `<name>.<namespace>.svc.cluster.local`. When `true` an alias is required. When `false` no alias is required. 
```
I'd just suggest trying to write the second sentence in present tense per the ISG's [guidance on verb tense](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense):
```suggestion
|Set to `true` to aggregate remote endpoint with local services. When `true` services are imported as `<name>.<namespace>.svc.cluster.local`. When `true` an alias is required. When `false` no alias is required. 
```

rh-tokeefe replied:
Made this edit.

skopacz1 commented on modules/ossm-federation-create-export.adoc:
```diff
@@ -11,6 +11,11 @@ You create an `ExportedServiceSet` resource to explicitly declare the services t
 
+[NOTE]
+====
+When you set the `importAsLocal` parameter to `true` to aggregate the remote endpoint with local services you must use an alias for the service. When you set the parameter `false` no alias is required.
```
Up to you but I think a comma here would be a helpful pause for the reader:
```suggestion
When you set the `importAsLocal` parameter to `true` to aggregate the remote endpoint with local services, you must use an alias for the service. When you set the parameter `false` no alias is required.
```

rh-tokeefe replied:
Made this edit.

skopacz1 reviewed (commented):
Left a couple of suggestions but otherwise LGTM!

skopacz1 commented:
/remove-label peer-review-needed
/remove-label peer-review-in-progress

/label peer-review-done

rh-tokeefe commented:
@skopacz1 thank you for the review. I've rebased the PR, made the edits, then pushed and squashed. The layered products are still getting acclimated to using the OCP workflow - they used to be done separately. I believe the next step is to add the merge review label. Is that something the peer reviewer does or should I do it?

skopacz1 commented:
@rh-tokeefe if you've completed all other reviews and are ready for this to be merged, yes you can add the merge-review-needed label. This is something you would do to ensure that it's only merged when you're ready for it to be

rh-tokeefe commented:
/label merge-review-needed

adellape commented on modules/ossm-federation-create-export.adoc:
```diff
@@ -11,6 +11,11 @@ You create an `ExportedServiceSet` resource to explicitly declare the services t
 
 Services are exported as `<export-name>.<export-namespace>.svc.<ServiceMeshPeer.name>-exports.local` and will automatically route to the target service.  This is the name by which the exported service is known in the exporting mesh. When the ingress gateway receives a request destined for this name, it will be routed to the actual service being exported. For example, if a service named `ratings.r ...
```
This comment is out of scope for this PR, but I noticed throughout the `ossm-federation` assembly, there are a lot of user-replaceable values that aren't following [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#command-syntax-for-replaceable-values) (e.g., all lowercase, `_` between multiple words, always marked up with backticks).

Would recommend doing a larger sweep in a follow-up PR for this types of issues.
 
```suggestion
Services are exported as `<export_name>.<export_namespace>.svc.<servicemeshpeer_name>-exports.local` and will automatically route to the target service.  This is the name by which the exported service is known in the exporting mesh. When the ingress gateway receives a request destined for this name, it will be routed to the actual service being exported. For example, if a service named `ratings.red-mesh-bookinfo` is exported to `green-mesh` as `ratings.bookinfo`, the service will be exported under the name `ratings.bookinfo.svc.green-mesh-exports.local`, and traffic received by the ingress gateway for that hostname will be routed to the `ratings.red-mesh-bookinfo` service.
```
(Just an example, not sure if `.name` is special for `ServiceMeshPeer` objects, like if it should be `<servicemeshpeer>.name`)

rh-tokeefe replied:
Created OSSM-5966 and assigned it to myself so this is not lost.

rh-tokeefe commented on modules/ossm-federation-create-export.adoc:
```diff
@@ -11,6 +11,11 @@ You create an `ExportedServiceSet` resource to explicitly declare the services t
 
+[NOTE]
+====
+When you set the `importAsLocal` parameter to `true` to aggregate the remote endpoint with local services, you must use an alias for the service. When you set the parameter `false` no alias is required.
```
Done

rh-tokeefe commented on modules/ossm-federation-config-import.adoc:
```diff
@@ -77,7 +77,7 @@ spec:
   - type: NameSelector
     importAsLocal:
-|Set to `true` to aggregate remote endpoint with local services. When `true`, services will be imported as `<name>.<namespace>.svc.cluster.local`
+|Set to `true` to aggregate remote endpoint with local services. When `true` services are imported as `<name>.<namespace>.svc.cluster.local`. When `true` an alias is required. When `false` no alias is required. 
```
Done

adellape commented:
/cherrypick enterprise-4.15

adellape commented:
/cherrypick enterprise-4.14

adellape commented:
/cherrypick enterprise-4.12
